### PR TITLE
Fix for  "RsWithType overrides status code"

### PR DESCRIPTION
**Pull request by ekondrashev**
Please review this PR fixing "RsWithType overrides status code" #298

The fix actually removes passing HTTP OK status code.

### CODE REVIEW

**Comment by davvd** [comment]
@ekondrashev Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@carlosmiranda it's yours, please review

File: src/main/java/org/takes/rs/RsWithType.java [hunk 1]
```diff
@@ -53,7 +52,7 @@ public RsWithType(final Response res, final CharSequence type) {
         super(
             new RsWithHeader(
                 new RsWithoutHeader(
-                    new RsWithStatus(res, HttpURLConnection.HTTP_OK),
+                    res,
```
Comment (carlosmiranda) [inline src/main/java/org/takes/rs/RsWithType.java]: @ekondrashev These two arguments in line 55 and 56 can be inlined in line 54.

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 2]
```diff
@@ -52,11 +57,41 @@ public void replaceTypeToResponse() throws IOException {
                 )
             ).print(),
             Matchers.equalTo(
-                Joiner.on("\r\n").join(
-                    "HTTP/1.1 200 OK",
-                    String.format("Content-Type: %s", type),
+                Joiner.on(CRLF).join(
+                        "HTTP/1.1 200 OK",
```
Comment (carlosmiranda) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @ekondrashev Too much indentation for lines 61-64, move them back four spaces.

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 3]
```diff
@@ -52,11 +57,41 @@ public void replaceTypeToResponse() throws IOException {
...
+            )
+        );
+    }
+
+    /**
+     * RsWithType can not replace response code.
+     * @throws java.io.IOException If a problem occurs.
+     */
+    @Test
+    public void notReplacesResponseCode() throws IOException {
```
Comment (carlosmiranda) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @ekondrashev Let's name this method `cannotReplaceResponseCode`. Also, let's declare `Exception` being thrown here, as per the practice [described here](http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html?expand=1#test-method-names).

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 4]
```diff
@@ -52,11 +57,41 @@ public void replaceTypeToResponse() throws IOException {
...
+        final int code = 500;
```
Comment (carlosmiranda) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @ekondrashev No need for this variable, we're only using it once.

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 5]
```diff
@@ -52,11 +57,41 @@ public void replaceTypeToResponse() throws IOException {
...
+        final String body = "Error!";
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new RsWithType(
+                    new RsWithBody(
+                        new RsWithStatus(code),
```
Comment (carlosmiranda) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @ekondrashev We should use constant `HttpURLConnection.HTTP_INTERNAL_ERROR` [(see javadoc)](http://docs.oracle.com/javase/7/docs/api/java/net/HttpURLConnection.html#HTTP_INTERNAL_ERROR).

**Comment by carlosmiranda** [comment]
@ekondrashev Some comments above.

**Comment by ekondrashev** [comment]
@carlosmiranda Done with your comments, please take a look if you find it ok.

**Comment by carlosmiranda** [comment]
@rultor merge please

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka please, let us know what do you think about this ticket, according to our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by ypshenychka** [comment]
@davvd Quality is good.

**Comment by davvd** [comment]
> @davvd Quality is good.

@ypshenychka thanks, got it, great to see high-quality tickets here

**Comment by davvd** [comment]
@carlosmiranda just added 10 mins to @ypshenychka (for QA), payment ID is `57722442`... 10 mins added to @yegor256 account (our architect), payment ID is `000-296dd5a5`, since quality is good here... **17 mins** was added to your account, many thanks for your contribution! The task took 57722453.... you're getting extra minutes here (c=2)... +17 added to your rating, at the moment it is: [+4112](http://www.netbout.com/b/35023?open=rating)

**Pull request merged**